src/ows/src/main/java/org/geoserver/ows/util/KvpUtils.java

```diff
@@ -345,12 +345,15 @@ public static String clean(String raw) {
      */
     public static KvpMap<String, String> toStringKVP(Map<String, ?> kvp) {
         KvpMap<String, String> result = new KvpMap<>();
-        for (Map.Entry<String, ?> entry : kvp.entrySet()) {
-            if (entry.getValue() instanceof String || entry.getValue() == null) {
-                result.put(entry.getKey(), (String) entry.getValue());
-            } else {
-                throw new IllegalArgumentException(
-                        "Found key with multiple values, this is unexpected: " + entry.getKey());
+        if (kvp != null) {
```

Comment:
So if the input is null, the result is empty but not null. Is this expected? (not sure really, I'm thinking out loud)

Reply:
Mmm.. A similar method was actually returning null indeed.
I have updated it right now to return null too.

Reply:
Thanks! Good to go